cpp/src/arrow/compute/kernels/scalar_arithmetic.cc

```diff
@@ -309,6 +348,21 @@ std::shared_ptr<ScalarFunction> MakeArithmeticFunctionNotNull(std::string name,
   return func;
 }
 
+template <typename Op>
+std::shared_ptr<ScalarFunction> MakeScalarArithmeticFunction(std::string name,
+                                                             const FunctionDoc* doc) {
+  auto func = std::make_shared<ScalarFunction>(name, Arity::Unary(), doc);
+  // 8-bit signed integer
+  // ArrayKernelExec exec = ScalarUnary<Int8Type, Int8Type, Op>::Exec;
+  // DCHECK_OK(func->AddKernel({int8()}, int8(), exec));
+
+  for (const auto& ty : NumericTypes()) {
```

Comment:
This will generate kernels for the unsigned integer types as well.
```suggestion
  for (const auto& ty : FlattenVectors({SignedIntTypes(), FloatingPointTypes()})) {
```